proparse/src/main/java/org/prorefactor/treeparser/ParseUnit.java

```diff
@@ -126,29 +116,45 @@ public File getFile() {
 
   public TokenStream lex() throws RefactorException {
     LOGGER.trace("Entering ParseUnit#lex()");
-    DoParse doParse = new DoParse(session, file.getPath());
     try {
-      doParse.doParse(true);
-    } catch (ANTLRException | IOException caught) {
+      ProgressLexer lexer = new ProgressLexer(session, file.getPath());
+      TokenStream stream = lexer.getANTLR2TokenStream(false);
+      try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1141)